src/transformers/models/wav2vec2/modeling_wav2vec2.py

```diff
@@ -893,8 +1066,38 @@ def forward(
 
         logits = self.lm_head(hidden_states)
 
+        loss = None
+        if labels is not None:
+
+            # retrieve loss input_lengths from attention_mask
+            attention_mask = (
+                attention_mask if attention_mask is not None else torch.ones_like(input_values, dtype=torch.long)
+            )
+            input_lengths = self._get_feat_extract_output_lengths(attention_mask.sum(-1))
+
+            # assuming that padded tokens are filled with -100
+            # when not being attended to
+            labels_mask = labels >= 0
+            target_lengths = labels_mask.sum(-1)
+            flattened_targets = labels.masked_select(labels_mask)
+
+            log_probs = F.log_softmax(logits, dim=-1).transpose(0, 1)
+
+            with torch.backends.cudnn.flags(enabled=False):
+                loss = F.ctc_loss(
+                    log_probs,
+                    flattened_targets,
+                    input_lengths,
+                    target_lengths,
+                    blank=self.config.pad_token_id,
+                    reduction="sum",
+                    zero_infinity=False,
```

Comment:
Ah! I would expose this parameter to the user if possible or set this to True by default, there are a few cases where CTC can throw a infinite loss at scale, and you probably want to ignore it